FirebaseDatabase/Sources/Realtime/FWebSocketConnection.m

```diff
@@ -79,12 +90,39 @@ - (id)initWith:(FRepoInfo *)repoInfo
 
         NSURLRequest *req = [[NSURLRequest alloc]
             initWithURL:[[NSURL alloc] initWithString:connectionUrl]];
+#if TARGET_OS_WATCH
```

Comment:
Did you take a look at what it would take to have a Watch-specific implementation of FWebSocketConnection? I am a bit worried about maintaining sane code coverage numbers for this file. 

Another suggestion would be to create an abstraction on top of `NSURLSessionWebSocketTask` and `FSRWebSocket`. That might be more straightforward and would cut most changes to this file.

Reply:
I did briefly and thought this was a reasonable tradeoff. Options + reasoning below...

1. Selecting files to include/exclude based on package manager.  This doesn't really work well, and would be overly complex.
2. Using separate files to contain each implementation and use `#ifdef` around the whole file. I was worried here about having implementation files with different naming and issues that may come up with it. Also with code duplication.
3. Abstraction on top. Thought about this but didn't seem like it was worth the effort, plus eventually we would want this implementation to take over for SocketRocket so it'd leave more work in the future to revert it.

Re: code coverage, the only reference to this class in tests is https://github.com/firebase/firebase-ios-sdk/blob/14764b8d60a6ad023d8fa5b7f81d42378d92e6fe/FirebaseDatabase/Tests/Unit/FUtilitiesTest.m#L109 so I don't think it's overly tested anyways (for better or for worse). 

The last time this file was modified outside of adding tvOS support or fixing imports (unrelated to functionality) was in June 2017 https://github.com/firebase/firebase-ios-sdk/commit/9917869191ecc00f6ac060fabe3cd4470bebbfab#diff-50e445e5405c9e69557c81053b2ada15120eda819c93ca059d14facffa892951 so I wasn't overly worried about it being modified often and becoming harder to maintain.

If I were to do anything, it would be create an abstraction between the 2 but I'd also worry about modifying the iOS code since it isn't well tested and hasn't been changed in a very long time. Let me know what you think.

Reply:
Sounds fair. It didn't occur to me that including the source and not using it would still be challenging.